packages/mask/background/services/helper/r2d2Fetch.ts

```diff
@@ -1,66 +1,66 @@
-    attemptUntil,
-    fetchImageViaDOM,
-    isIPFS_CID,
-    resolveCrossOriginURL,
-    resolveIPFS_URL,
-} from '@masknet/web3-shared-base'
-
 /* cspell:disable */
-const R2D2_ROOT_URL = 'r2d2.to'
-
-enum R2d2Workers {
-    alchemy = 'alchemy-proxy',
-    opensea = 'opensea-proxy',
-    gitcoin = 'gitcoin-agent',
-    coinMarketCap = 'coinmarketcap-agent',
-    goPlusLabs = 'gopluslabs',
-    coingecko = 'coingecko-agent',
-}
+import { attemptUntil, fetchImageViaDOM } from '@masknet/web3-shared-base'
 
-enum AlchemyProxies {
-    eth = 'eth-mainnet',
-    eth_io = 'eth-mainnet-io',
-    polygon = 'polygon-mainnet',
-    flow = 'flow-mainnet',
-}
+const R2D2_ROOT_URL = 'r2d2.to'
 
-const IPFS_NODE_URLS = ['ipfs.infura.io']
+const INFURA_IPFS_ROOT_URL = 'ipfs.infura.io'
 
 const HOTFIX_RPC_URLS = [
-    'infura.io',
+    'mainnet.infura.io',
+    'ropsten.infura.io',
+    'rinkeby.infura.io',
+    'kovan.infura.io',
+    'goerli.infura.io',
+    'polygon-mainnet.infura.io',
+    'polygon-mumbai.infura.io',
     'quiknode.pro',
     'binance.org',
-    'aurora.dev',
-    'onfinality.io',
-    'arbitrum.io',
-    'celo.org',
-    'ftm.tools',
-    'gnosischain.com',
-    'avax.network',
-    'optimism.io',
+    'polygon-rpc.com',
+    'mainnet.aurora.dev',
+    'testnet.aurora.dev',
+    'astar.api.onfinality.io',
+    'rpc.astar.network',
+    'arb1.arbitrum.io',
+    'rinkeby.arbitrum.io',
+    'forno.celo.org',
+    'rpc.ftm.tools',
+    'rpc.gnosischain.com',
+    'api.avax.network',
+    'mainnet.optimism.io',
+    'kovan.optimism.io',
+    'goerli.optimism.io',
     'harmony.one',
     'hmny.io',
-    'hermesdefi.io',
-    'pokt.network',
-    'confluxrpc.com',
+    'rpc.hermesdefi.io',
+    'gateway.pokt.network',
+    'evm.confluxrpc.com',
 ]
 
-const AlchemyMatchers: Array<[string, AlchemyProxies]> = [
-    ['https://eth-mainnet.alchemyapi.io', AlchemyProxies.eth_io],
-    ['https://eth-mainnet.g.alchemy.com', AlchemyProxies.eth],
-    ['https://polygon-mainnet.g.alchemy.com', AlchemyProxies.polygon],
-    ['https://flow-mainnet.g.alchemy.com', AlchemyProxies.flow],
-]
+const { fetch: originalFetch } = globalThis
 
-const WorkerMatchers: Array<[string, R2d2Workers]> = [
-    ['https://api.opensea.io', R2d2Workers.opensea],
-    ['https://web-api.coinmarketcap.com', R2d2Workers.coinMarketCap],
-    ['https://api.gopluslabs.io', R2d2Workers.goPlusLabs],
-    ['https://api.coingecko.com', R2d2Workers.coingecko],
-]
+async function squashedFetch(request: RequestInfo, init?: RequestInit): Promise<Response> {
+    const cache = await caches.open('r2d2')
+    const hit = await cache.match(request)
+    if (hit) {
+        console.log('DEBUG: hit')
+        console.log({
+            request,
+            hit,
+        })
+        return hit
+    }
 
-const { fetch: originalFetch } = globalThis
+    const response = await originalFetch(request, init)
+    if (response.ok && response.status === 200) {
+        await cache.put(request, response)
```

Comment:
Add a try-catch on it, if it is unavailable (maybe on iOS?), it will not block all of our traffic.